### PR TITLE
MacOS compatibility 

### PR DESCRIPTION
**Pull request by ubgk**
This PR introduces MacOS compatibility by using an alternative of the Linux-specific `pthread_setname_np` which has a different signature for MacOS.

### CODE REVIEW

File: .github/workflows/bazel.yml [hunk 1]
```diff
@@ -44,7 +56,10 @@ jobs:
 
             - name: "Report test coverage"
               run: |
-                  tools/bazelisk coverage --combined_report=lcov --instrument_test_targets //...
+                  tools/bazelisk coverage --combined_report=lcov --instrument_test_targets --test_verbose_timeout_warnings //...
+
+            - name: Setup upterm session
+              uses: lhotari/action-upterm@v1
```
Comment (stephane-caron) [inline .github/workflows/bazel.yml]: This can come in handy, but we'll want to remove it otherwise tests won't terminate by themselves, no?

**Review by stephane-caron — COMMENTED** [review]
LGTM, except the upterm session which doesn't seem to terminate (unless we actually log into the upterm and close the terminal).

File: .github/workflows/bazel.yml (hunk 1 above)
Comment (ubgk) [reply .github/workflows/bazel.yml]: I'm not done with the PR yet, upterm's there to debug for now − we should indeed remove it when we're done. For some reason the MacOS job fails at `Submit report to Coveralls`: [here](https://github.com/boragokbakan/mpacklog/actions/runs/4744791127/jobs/8426188109)

Based on logs I suspect the previous step is not creating the coverage report properly on MacOS.

Comment (stephane-caron) [reply .github/workflows/bazel.yml]: Gotcha, you used the draft status properly :+1: For coveralls, we can execute that step on Linux only.

Comment (ubgk) [reply .github/workflows/bazel.yml]: We're all green now then! :)

**Comment by stephane-caron** [comment]
LGTM, thanks :smiley:

**Pull request merged**